presets/ragengine/main.py

```diff
@@ -90,23 +95,47 @@ async def list_documents_in_index(index_name: str):
     my_index          | my_index          | my_index
     my index          | my%20index        | my index
     index/name        | index%2Fname      | index/name
-    index@name        | index%40name      | index@name
-    index#1           | index%231         | index#1
-    index?query       | index%3Fquery     | index?query
     """
     try:
         # Decode the index_name in case it was URL-encoded by the client
         decoded_index_name = unquote(index_name)
-        documents = await rag_ops.list_documents_in_index(decoded_index_name)
-        return ListDocumentsResponse(documents=documents)
+        documents = await rag_ops.list_documents_in_index(
+            index_name=decoded_index_name,
+            limit=limit+1, # Fetch one extra to check for more results
```

Comment:
I don't understand the +1 part here.

Reply:
Yes we grab one more than the limit to see if there are more possible documents that can be returned - it is to set the has_more and next_offset fields